Src/CSharpier/SyntaxPrinter/UsingDirectives.cs

```diff
@@ -0,0 +1,259 @@
+        var aliasNameUsings = new List<UsingData>();
+        var regularUsings = new List<UsingData>();
+        var staticUsings = new List<UsingData>();
+        var aliasUsings = new List<UsingData>();
+        var directiveGroup = new List<UsingData>();
+        var ifCount = 0;
+        var isFirst = true;
+
+        foreach (var usingDirective in usings)
+        {
+            var openIf = ifCount > 0;
+            foreach (var directive in usingDirective.GetLeadingTrivia().Where(o => o.IsDirective))
+            {
+                if (directive.RawSyntaxKind() is SyntaxKind.IfDirectiveTrivia)
+                {
+                    ifCount++;
+                }
+                else if (directive.RawSyntaxKind() is SyntaxKind.EndIfDirectiveTrivia)
+                {
+                    ifCount--;
+                }
+            }
+
+            if (ifCount > 0)
+            {
+                directiveGroup.Add(
+                    new UsingData
+                    {
+                        Using = usingDirective,
+                        LeadingTrivia = PrintLeadingTrivia(usingDirective)
+                    }
+                );
+            }
+            else
+            {
+                if (openIf)
+                {
+                    directiveGroup.Add(
+                        new UsingData { LeadingTrivia = PrintLeadingTrivia(usingDirective) }
+                    );
+                }
+
+                var usingData = new UsingData
+                {
+                    Using = usingDirective,
+                    LeadingTrivia = !openIf ? PrintLeadingTrivia(usingDirective) : Doc.Null
+                };
+
+                if (usingDirective.GlobalKeyword.RawSyntaxKind() != SyntaxKind.None)
+                {
+                    globalUsings.Add(usingData);
+                }
+                else if (usingDirective.StaticKeyword.RawSyntaxKind() != SyntaxKind.None)
+                {
+                    staticUsings.Add(usingData);
+                }
+                else if (usingDirective.Alias is not null)
+                {
+                    aliasUsings.Add(usingData);
+                }
+                else if (usingDirective.Name is AliasQualifiedNameSyntax)
+                {
+                    aliasNameUsings.Add(usingData);
+                }
+                else if (usingDirective.Name is not null && IsSystemName(usingDirective.Name))
+                {
+                    systemUsings.Add(usingData);
+                }
+                else
+                {
+                    regularUsings.Add(usingData);
+                }
+            }
+        }
+
+        yield return globalUsings.OrderBy(o => o.Using!, Comparer).ToList();
+        yield return systemUsings.OrderBy(o => o.Using!, Comparer).ToList();
+        yield return aliasNameUsings.OrderBy(o => o.Using!, Comparer).ToList();
+        yield return regularUsings.OrderBy(o => o.Using!, Comparer).ToList();
```

Comment:
I suspect if you just do `.OrderBy(o => "" + o.Using?.Alias.ToFullString() + o.Using?.Name.ToFullString())`, you won't need that comparer.
Though it's long enough that it might be worth extracting  to a helper method, and it might be slower because of the string concat, so :shrug: